prow/plugins/golint/golint_test.go

```diff
@@ -240,3 +241,136 @@ func TestAddedLines(t *testing.T) {
+		expectedModifiedFiles map[string]string
+	}{
+		{
+			name: "modified files include vendor file",
+			gh: &ghc{
+				changes: []github.PullRequestChange{
+					{
+						Filename: "qux.go",
+						Patch:    "@@ -0,0 +1,5 @@\n+package bar\n+\n+func Qux() error {\n+   return nil\n+}",
+					},
+					{
+						Filename: "vendor/foo/bar.go",
+						Patch:    "@@ -0,0 +1,5 @@\n+package bar\n+\n+func Qux2() error {\n+   return nil\n+}",
+					},
+				},
+			},
+			expectedModifiedFiles: map[string]string{
+				"qux.go": "@@ -0,0 +1,5 @@\n+package bar\n+\n+func Qux() error {\n+   return nil\n+}",
+			},
+		},
+		{
+			name: "modified files include non go file",
+			gh: &ghc{
+				changes: []github.PullRequestChange{
+					{
+						Filename: "qux.go",
+						Patch:    "@@ -0,0 +1,5 @@\n+package bar\n+\n+func Qux() error {\n+   return nil\n+}",
+					},
+					{
+						Filename: "foo.come",
+						Patch:    "@@ -0,0 +1,5 @@\n+package bar\n+\n+func Qux2() error {\n+   return nil\n+}",
+					},
+				},
+			},
+			expectedModifiedFiles: map[string]string{
+				"qux.go": "@@ -0,0 +1,5 @@\n+package bar\n+\n+func Qux() error {\n+   return nil\n+}",
+			},
+		},
+		{
+			name: "modified files include generated file",
+			gh: &ghc{
+				genfile: []byte("file-prefix zz_generated"),
+				changes: []github.PullRequestChange{
+					{
+						Filename: "qux.go",
+						Patch:    "@@ -0,0 +1,5 @@\n+package bar\n+\n+func Qux() error {\n+   return nil\n+}",
+					},
+					{
+						Filename: "zz_generated.wowza.go",
+						Patch:    "@@ -0,0 +1,5 @@\n+package bar\n+\n+func Qux2() error {\n+   return nil\n+}",
+					},
+				},
+			},
+			expectedModifiedFiles: map[string]string{
+				"qux.go": "@@ -0,0 +1,5 @@\n+package bar\n+\n+func Qux() error {\n+   return nil\n+}",
+			},
+		},
+		{
+			name: "modified files include removed file",
+			gh: &ghc{
+				changes: []github.PullRequestChange{
+					{
+						Filename: "qux.go",
+						Patch:    "@@ -0,0 +1,5 @@\n+package bar\n+\n+func Qux() error {\n+   return nil\n+}",
+					},
+					{
+						Filename: "bar.go",
+						Status:   github.PullRequestFileRemoved,
+						Patch:    "@@ -0,0 +1,5 @@\n+package bar\n+\n+func Qux2() error {\n+   return nil\n+}",
+					},
+				},
+			},
+			expectedModifiedFiles: map[string]string{
+				"qux.go": "@@ -0,0 +1,5 @@\n+package bar\n+\n+func Qux() error {\n+   return nil\n+}",
+			},
+		},
+		{
+			name: "modified files include renamed file",
+			gh: &ghc{
+				changes: []github.PullRequestChange{
+					{
+						Filename: "qux.go",
+						Patch:    "@@ -0,0 +1,5 @@\n+package bar\n+\n+func Qux() error {\n+   return nil\n+}",
+					},
+					{
+						Filename: "bar.go",
+						Status:   github.PullRequestFileRenamed,
+						Patch:    "@@ -0,0 +1,5 @@\n+package bar\n+\n+func Qux2() error {\n+   return nil\n+}",
+					},
+				},
+			},
+			expectedModifiedFiles: map[string]string{
+				"qux.go": "@@ -0,0 +1,5 @@\n+package bar\n+\n+func Qux() error {\n+   return nil\n+}",
+			},
+		},
+	}
```

Comment:
Add two more test cases: one where more than a single modified file is expected, and another one where none of the changes are expected.